Inconsistently generating MACI keypairs

After connecting with a wallet which I've already registered in MACI with, I am prompted to "sign up" in the UI. When I try to "sign up", the UI throws an error toast notification (seemingly because the tx would revert).

This appears to mean that we're not generating deterministic MACI keys based off the ETH address?
Here's the address I was using: `0x1b6ec1327e70075CC43EDF08c5F8230A91d9a98f`

--- 

Round has completed
1. I navigate to Stats page (https://maci-rpgf.vercel.app/stats)
2. I click "Sign up"
3. Error toast pops

Solution:
I think we should just not display the "Sign up" prompt if the round is already over

Video:
https://github.com/privacy-scaling-explorations/maci-rpgf/assets/8097623/8985e492-9feb-4efd-a1da-905f03cc1e76

Screenshot:
![Image 2024-04-26 at 13 34 40](https://github.com/privacy-scaling-explorations/maci-rpgf/assets/8097623/aabc34c9-26ed-4483-b7e5-fd4e9e3d9b19)

Console errors:
```
VM43:1 MetaMask - RPC Error: Internal JSON-RPC error. {code: -32603, message: 'Internal JSON-RPC error.', data: {…}}code: -32603data: {code: 3, message: 'execution reverted', data: '0x3a81d6fc', cause: null}message: "Internal JSON-RPC error."[[Prototype]]: Object
(anonymous) @ VM43:1
(anonymous) @ VM43:1
(anonymous) @ VM43:1
(anonymous) @ VM43:1
m @ VM43:1
p @ VM43:1
(anonymous) @ VM43:1
s @ VM43:1
Promise.then (async)
u @ VM43:1
(anonymous) @ VM43:1
m @ VM43:1
d @ VM43:1
(anonymous) @ VM43:1
(anonymous) @ VM43:1
m @ VM43:1
c @ VM43:1
handle @ VM43:1
_rpcRequest @ VM43:1
(anonymous) @ VM43:1
request @ VM43:1
#U @ _app-3c0f77d989df5d48.js:312
_send @ _app-3c0f77d989df5d48.js:312
(anonymous) @ _app-3c0f77d989df5d48.js:312
(anonymous) @ _app-3c0f77d989df5d48.js:312
setTimeout (async)
#e7 @ _app-3c0f77d989df5d48.js:312
send @ _app-3c0f77d989df5d48.js:312
send @ _app-3c0f77d989df5d48.js:312
await in send (async)
_perform @ _app-3c0f77d989df5d48.js:312
#eV @ _app-3c0f77d989df5d48.js:312
estimateGas @ _app-3c0f77d989df5d48.js:312
(anonymous) @ _app-3c0f77d989df5d48.js:312
sendUncheckedTransaction @ _app-3c0f77d989df5d48.js:312
sendTransaction @ _app-3c0f77d989df5d48.js:312
await in sendTransaction (async)
b @ _app-3c0f77d989df5d48.js:445
await in b (async)
signUp @ _app-3c0f77d989df5d48.js:445
i @ _app-3c0f77d989df5d48.js:299
await in i (async)
(anonymous) @ _app-3c0f77d989df5d48.js:312
(anonymous) @ 4617-1df04db64fe694a5.js:1
eU @ framework-e182ded6da661f6f.js:9
eH @ framework-e182ded6da661f6f.js:9
(anonymous) @ framework-e182ded6da661f6f.js:9
re @ framework-e182ded6da661f6f.js:9
rn @ framework-e182ded6da661f6f.js:9
(anonymous) @ framework-e182ded6da661f6f.js:9
oN @ framework-e182ded6da661f6f.js:9
eF @ framework-e182ded6da661f6f.js:9
ro @ framework-e182ded6da661f6f.js:9
nU @ framework-e182ded6da661f6f.js:9
nD @ framework-e182ded6da661f6f.js:9
_app-3c0f77d989df5d48.js:312 error happened: Error: [✗] execution reverted (unknown custom error) (action="estimateGas", data="0x3a81d6fc", reason=null, transaction={ "data": "0x3364120a1f3cd28b39fef1a0c2682dc5cc235a4b124512b30cd48581ba207c5aa78f64451116c1f9eab4c8047f5ab1d0dd8718143866a7e64fab541d1c5683934d7e43f1000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000000209029ce84bc4836aa74e6a61508436818dc00082897aa98abe303ae374d96885400000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000", "from": "0x1b6ec1327e70075CC43EDF08c5F8230A91d9a98f", "to": "0xD18Ca45b6cC1f409380731C40551BD66932046c3" }, invocation=null, revert=null, code=CALL_EXCEPTION, version=6.12.0)
```

--- a/src/pages/api/auth/[...nextauth].ts
+++ b/src/pages/api/auth/[...nextauth].ts
@@ -4,7 +4,6 @@ import { genKeyPair } from "maci-cli/sdk";
 import type { NextApiRequest, NextApiResponse } from "next";
 import NextAuth, { type NextAuthOptions } from "next-auth";
 import CredentialsProvider from "next-auth/providers/credentials";
-import { getCsrfToken } from "next-auth/react";
 import { SiweMessage } from "siwe";
 import { db } from "~/server/db";
 
@@ -21,8 +20,6 @@ export function getAuthOptions(req: IncomingMessage): NextAuthOptions {
             process.env.NEXTAUTH_URL ?? process.env.VERCEL_URL,
           ).origin;
 
-          console.log({ nextAuthUrl });
-
           if (!nextAuthUrl) {
             return null;
           }
@@ -32,14 +29,8 @@ export function getAuthOptions(req: IncomingMessage): NextAuthOptions {
             return null;
           }
 
-          if (
-            siwe.resources?.[0] !==
-            (await getCsrfToken({ req: { headers: req.headers } }))
-          ) {
-            return null;
-          }
-
           await siwe.verify({ signature: credentials?.signature ?? "" });
+          
           const keypair = genKeyPair({
             seed: credentials?.signature
               ? BigInt(credentials.signature)

--- a/src/providers/RainbowKitSiweNextAuthProvider.tsx
+++ b/src/providers/RainbowKitSiweNextAuthProvider.tsx
@@ -34,14 +34,13 @@ export function RainbowKitSiweNextAuthProvider({
   const adapter = useMemo(
     () =>
       createAuthenticationAdapter({
-        createMessage: ({ address, chainId, nonce }) => {
+        createMessage: ({ address, chainId }) => {
           const defaultConfigurableOptions: ConfigurableMessageOptions = {
             domain: window.location.host,
             statement: "Sign in with Ethereum to the app.",
             uri: window.location.origin,
             version: "1",
             nonce: address,
-            resources: [nonce],
             issuedAt: new Date(1).toISOString(),
           };
 

